apps/src/lib/client/tx.rs

```diff
@@ -686,9 +693,19 @@ async fn submit_tx(
             args.gas_limit.clone(),
             tx,
         );
-        match broadcast_tx(args.ledger_address.clone(), tx, keypair).await {
-            Ok(result) => (ctx, result.initialized_accounts),
-            Err(err) => {
+        // Either broadcast or submit transaction and collect result into
+        // sum type
+        let result = if args.broadcast_only {
+            Err(broadcast_tx(args.ledger_address.clone(), tx, keypair).await)
```

Comment:
I think the `Err` usage is a bit confusing here, you can probably use `itertools::Either` instead

Reply:
Thanks! Done - now using `itertools::Either` instead.